dbms/src/Storages/StorageMergeTree.cpp

```diff
@@ -191,6 +193,42 @@ void StorageMergeTree::rename(const String & new_path_to_db, const String & /*ne
 }
 
 
+std::vector<MergeTreeData::AlterDataPartTransactionPtr> StorageMergeTree::prepareAlterTransactions(
+    const ColumnsDescription & new_columns, const IndicesDescription & new_indices, const size_t thread_pool_size)
+{
+    ThreadPool thread_pool(thread_pool_size);
```

Comment:
Let's also cap the thread pool size to the number of data parts.

Reply:
done